pandas/tests/extension/decimal/test_decimal.py

```diff
@@ -380,6 +380,16 @@ def test_divmod_array(reverse, expected_div, expected_mod):
     tm.assert_extension_array_equal(mod, expected_mod)
 
 
+def test_na_count():
```

Comment:
It'd be good to make this a base test. http://pandas-docs.github.io/pandas-docs-travis/development/extending.html#testing-extension-arrays

It can go in `pandas/tests/extension/base/methods.py`, and it can use the `data_missing` fixture.

Reply:
hmm. Why doesn't `BaseMethodsTests.test_count` catch this bug then?

Reply:
`Dataframe.count` doesn't delegate to `series.count` (for perf I guess) so their behavior can diverge. and the test suite doesn't test that, hence this bug and possibly many others for other functions (Not volunteering). 

I'll add a `series_count` test for now.